src/ucai/core/utils/callable_utils.py

```diff
@@ -0,0 +1,285 @@
+import ast
+import inspect
+from textwrap import dedent, indent
+from typing import Any, Callable, Union, get_type_hints
+
+from ucai.core.utils.type_utils import python_type_to_sql_type
+
+FORBIDDEN_PARAMS = ['self', 'cls']
+
+def parse_docstring(docstring: str) -> dict[str, str]:
```

Comment:
I'd return a dataclass with `args`, `return`, and `raises` fields. Dict should not be used if you know the field names beforehand. In this particular case, we know `args`, `return`, and `raises` are always present.

Reply:
good call